registry/auth/basic/htpasswd.go

```diff
@@ -0,0 +1,54 @@
+package basic
+
+import (
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/csv"
+	"errors"
+	"os"
+)
+
+// ErrSHARequired - returned in error field of challenge when the htpasswd was not made using SHA1 algorithm.
+// 	            (SHA1 is considered obsolete but the alternative for htpasswd is MD5, or system crypt...)
+var ErrSHARequired = errors.New("htpasswd file must use SHA (htpasswd -s)")
+
+// HTPasswd - holds a path to a system .htpasswd file and the machinery to parse it.
+type HTPasswd struct {
+	path   string
+	reader *csv.Reader
+}
+
+// NewHTPasswd - Create a new HTPasswd with the given path to .htpasswd file.
+func NewHTPasswd(htpath string) *HTPasswd {
+	return &HTPasswd{path: htpath}
+}
+
+// AuthenticateUser - Check a given user:password credential against the receiving HTPasswd's file.
+func (htpasswd *HTPasswd) AuthenticateUser(user string, pwd string) (bool, error) {
+
+	// Hash the credential.
+	sha := sha1.New()
```

Comment:
Sure.   The full set of options supported by htpasswd are SHA, BCrypt, MD5 plaintext, and crypt().  SHA is a sensible default (although considered obsolete, it is not yet practically breakable).   MD5, plaintext, and system crypt() cannot be considered secure, but we could totally support and leave that concern to the user.   I'll take a look at the .htpasswd generated by each of the methods and update the branch.